components/concept/sync/src/main/java/mozilla/components/concept/sync/DeviceEvents.kt

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.sync
+
+/**
+ * Allows monitoring events targeted at the current device.
+ */
+interface DeviceEventsObserver {
+    fun onTabsReceived(tabs: List<TabData>, from: Device?)
```

Comment:
Shouldn't the `DeviceEventsObserver` observe `DeviceEvent`s. Seemed strange to have `onTabsReceveived` and also `DeviceEnvent.TabsReceived`.

Reply:
Ha, yeah. That is pretty weird. Changing.

Reply:
So, the intention here was to have two layers of event processing; an internal layer - for "system" events, like "please wipe your history storage", and for events that are handled downstream, such as "you got a new tab".

But, I didn't quite realize that vision anyway, and it's a future requirement. So let's keep this simple for now.

Reply:
Better example would be forcing logout remotely ("pls logout!"), or informing of new devices in the constellation. We already have some support for the latter.